### PR TITLE
Change all instances of recurring_event in the code to event

### LINKED ISSUE #173: Update Table: recurring_event (rename to event)
### Overview
We need to update the recurring_event table model (now the event table model) to utilize the practice area table instead of the outdated role table.

#### Details
The initial model issue is #14

Discussion leading to this change is here: https://github.com/hackforla/peopledepot/discussions/170

### Action Items
- [x] Update schema information
  - [x] add/remove given columns from [Entity Relationship Diagram (ERD)](https://lucid.app/lucidchart/ac2f3e81-00d2-4257-b1fc-266d7f0a4cbe/view)
  - [x] add/remove given columns from [PD: Table and Field definitions](https://docs.google.com/spreadsheets/d/1x_zZ8JLS2hO-zG0jUocOJmX16jh-DF5dccrd_OEGNZ0/edit#gid=1572339087)
  - [x] make additional schema changes (if needed)
  - [x] remove the ERD/SS update label after these changes are made.
     - Permission is needed to edit the ERD and table spreadsheet.
- [x] Rename Django model
- [x] Update existing Django model
- [ ] Write a test for the new relationships this model will have with other models (e.g., creating a user and assigning them a set of permissions on a project) if any.
- [x] Update API end point
- [x] Update API unit tests
- [ ] Document the endpoint in Swagger
- [ ] Take #88 out of the ice box

#### Columns to Remove
- [x]  must_roles - int[] (role_id)
- [x]  should_roles - int[] (role_id)
- [x]  could_roles - int[] (role_id)

#### Columns to Add
- [x]  must_attend - JSON[]
- [x]  should_attend - JSON[]
- [x]  could_attend - JSON[]

#### Additional Changes
- [x]  Change name from recurring_event to event
  - [x] in schema
  - [x] in django
- [x]  Add example JSON to the ERD somehow
     - This could be handled by adding a stickynote to the ERD, or maybe a link to a new tab in the spreadsheet.
JSON Example:
```
[{practice_area: engineering, permission_level: practiceLeadProject},{practice_area: pm, permission_level: practiceLeadProject}]
```
### Resources
- See [People Depot Resources wiki page](https://github.com/hackforla/peopledepot/wiki/Resources-and-Links) for links
  - ERD
  - Table and Field Definitions
  - API Endpoint

### PR DESCRIPTION
**Pull request by anandramakris**
Fixes #173 

### What changes did you make?

- Changed all instances of recurring_event to event outside of the migration files and the documentation
- Created new migration file for the change

### Why did you make the changes (we will use this info to test)?

- Model should support one-time events in addition to recurring events

### Screenshots of Proposed Changes Of The Website (if any, please do not screen shot code changes)]

<details>
<summary>Visuals before changes are applied</summary>

image

</details>

<details>
<summary>Visuals after changes are applied</summary>

image

</details>

### CODE REVIEW

File: app/core/api/serializers.py [hunk 1]
```diff
@@ -105,11 +105,11 @@ class Meta:
         )
 
 
-class RecurringEventSerializer(serializers.ModelSerializer):
+class EventSerializer(serializers.ModelSerializer):
     """Used to retrieve recurring_event info"""
```
Comment (AzaniaBG) [inline app/core/api/serializers.py:109]: The doc string still refers to "recurring_event"

**Review by AzaniaBG — COMMENTED** [review]
@anandramakris, nice work on this. It appears all instances of recurring_event have been changed to event. Also, thank you for pointing out that documentation will need to be changed. Specifically, the [add-models-and api-endpoints.md](https://github.com/hackforla/peopledepot/blob/main/docs/how-to/add-model-and-api-endpoints.md#add-the-model-in-django) still references the "recurring_event" model.

I left an inline comment where the doc string still refers to "recurring_event" (in `serializers.py`). Also, when I navigate to localhost:8000/admin/core/event, I see an error:"core_event" does not exist. I'm not sure if migrations need to be rolled back and run again, because I'm seeing an exception/error "raise InconsistentMigrationHistory" when I run `./scripts/buildrun.sh.` 

@fyliu: I believe these are the changes that need to be made:
1. change the doc string to reference "event"
2. fix the inconsistent migration history
Also, in the future, I suggest using a branch name that includes the issue number, instead of using the `main` branch.

**Comment by anandramakris** [comment]
I didn't get that error at all, thanks for noticing on your end.

**Comment by anandramakris** [comment]
Tested right now on my end if the migration error still occurs and it just says "No migrations to apply."

**Review by fyliu — CHANGES_REQUESTED** [review]
Thanks for working on this.

I see that you re-did the 0004_recurringevent.py migration into 0004_event.py migration. You should actually just leave the old migration files alone and let django generate a new migration file to do the deletions. You probably weighed the choices and it felt cleaner to do this. But rewriting migration files will break the database for the people have have the project deployed locally, where their databases were built using the old migration files. We don't have a publicly-deployed instance to worry about for now, but we should always add to migrations so that running instance can update to the new database schema.

Before public deployment, we will flatten all the existing migration file into one to reduce the number of them and to get rid of design redo changes like this one.

**Comment by fyliu** [comment]
@anandramakris yes, exactly. We should let it generate a new migration for the change.

**Comment by AzaniaBG** [comment]
@fyliu @anandramakris ~ I might be missing something, but when I click on the GH email link for the latest commit, I see the following message: "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository."
Screenshot 2023-10-25 at 6 10 16 AM

**Comment by AzaniaBG** [comment]
@anandramakris @fyliu ~ I requested changes. Anand, can you edit your commit message to add the changes you made using this template[this pull request template](https://github.com/hackforla/peopledepot/compare/29b4eddafdf911a0b2157563c26d270073c22cfa..b8a3766c61ca106e66c3d18ab6e995caa168dd84#diff-b2496e80299b8c3150b1944450bd81c622e04e13d15c411d291db0927d75fd6b)?

Also, the force commit has caused an issue that makes it difficult to find the commit on the branch. For example, I see a message that the commit does not belong to a branch on the repo when clicking the GH email commit link to get to your   latest commit. Also, when I navigate to issue #209, I see a message that the commits can't be found (see screenshots below). 

Screenshot 2023-10-25 at 6 34 26 AM

Screenshot 2023-10-25 at 6 31 16 AM

**Review by fyliu — CHANGES_REQUESTED** [review]
Looks good. The only thing is probably changing the short message of the commit to something other than "Fixed migration files". Something like "Rename recurring_event to event" is better. `git commit --amend` should let you edit the message. Then I guess you'll have to force push

And the PR template @AzaniaBG talked about if you want to copy it from the templates directory and replace the PR description with it.

What I did to test

- I deleted my data volume, built the previous commit, updated to your commit, and ran migrate. It applied the new 0018 migration without errors.

    ```bash
    docker-compose down -v

    # reset to previous commit

    ./scripts/buildrun.sh

    # reset to PR commit

    ./scripts/migrate.sh
    ```

- admin site and api look okay

- Grepping for "recurring" didn't show any more that need changing

    ```bash
    grep -nirs -I --exclude-dir=".git" "recurring" .
    ```

**Comment by anandramakris** [comment]
Have changed the commit message.

**Review by fyliu — APPROVED** [review]
Looks good. Thanks for making the changes.

**Pull request merged**